### ISSUE TITLE
OpenAPI Emitter: Response should respect @useRef 

### ISSUE THREAD

**Issue by mcchrish** [microsoft/typespec#3601]
### Clear and concise description of the problem

```
namespace Pet Service {
  @get
  op get(@path petId: string): CustomResponse;
}

@useRef("https://example.com/example.yml")
model CustomResponse {
  @statusCode
  status: 200;
}
```

Should compile to:

```yaml
      responses:
        '200':
          $ref: 'https://example.com/example.yml'
```

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://typespec.io/docs/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by markcowl** [comment]
p2

**Comment by sarangan12** [comment]
Playground [Link](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQrSGnJlc3TWGm9wZW5hcGkzxB4NCnVzaW5nIFR5cGVTcGVjLkh0dHA70RZSZXN00hZPcGVuQVBJxUdAc2VydmljZSh7DQogIHRpdGxlOiAiUGV0IFPGGSIsDQp9KQ0KbmFtZXNwYWNlIMQdxS5AZ2V0xAhvcCBnZXQoQHBhdGggcGV0SWQ6IHN0cmluZyk6IEN1c3RvbVJlc3BvbnNlOw0KfcV3dXNlUmVmKCLkAQlzOi8vZXhhbXBsZS5jb23JDHltbCIpDQptb2RlbM9I5wCBc3RhdHVzQ29kZcQPxg46IDIwMMRo&e=%40typespec%2Fopenapi3&options=%7B%7D)

**Comment by sarangan12** [comment]
Code Complete

**Comment by chrisradek** [comment]
Closed by #3894

**Comment by wingfaic** [comment]
Should $ref be emitted without siblings? (specifically the built in response descriptions which can't be removed)

Otherwise it'll cause this default spectral rule to fail https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules#no-ref-siblings.

Example
```
@useRef("https://example.com")
model Error {
  @statusCode _: 500;
}

@useRef("https://example.com")
model Response {}

@route("/get")
op get(): Response | Error;
```

0.59.0-dev.4 Currently outputs 
```
get:
  operationId: get
  parameters: []
  responses:
    '200':
      description: The request has succeeded.
      $ref: https://example.com
    '500':
      description: Server error
      $ref: https://example.com
```

But should be
```
get:
  operationId: get
  parameters: []
  responses:
    '200':
      $ref: https://example.com
    '500':
      $ref: https://example.com
```
